packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -916,35 +999,55 @@ class CpuProfileStore {
 
 class _CpuProfileTimelineTree {
   factory _CpuProfileTimelineTree.fromCpuSamples(
-    vm_service.CpuSamples cpuSamples,
-  ) {
-    final root = _CpuProfileTimelineTree._fromIndex(cpuSamples, kRootIndex);
+    vm_service.CpuSamples cpuSamples, {
+    bool buildCodeTree = false,
+  }) {
+    final root = _CpuProfileTimelineTree._fromIndex(
+      cpuSamples,
+      kRootIndex,
+      buildCodeTree,
+    );
     _CpuProfileTimelineTree current;
     // TODO(bkonyi): handle truncated?
-    for (final sample in cpuSamples.samples ?? []) {
+    for (final sample in cpuSamples.samples ?? <vm_service.CpuSample>[]) {
       current = root;
+      final stack = buildCodeTree ? sample.codeStack : sample.stack!;
       // Build an inclusive trie.
-      for (final index in sample.stack!.reversed) {
+      for (final index in stack.reversed) {
         current = current._getChild(index);
       }
       _timelineTreeExpando[sample] = current;
     }
     return root;
   }
 
-  _CpuProfileTimelineTree._fromIndex(this.samples, this.index);
+  _CpuProfileTimelineTree._fromIndex(this.samples, this.index, this.isCodeTree);
 
   static final _timelineTreeExpando = Expando<_CpuProfileTimelineTree>();
   static const kRootIndex = -1;
   static const kNoFrameId = -1;
   final vm_service.CpuSamples samples;
   final int index;
+  final bool isCodeTree;
   int frameId = kNoFrameId;
 
-  String? get name => samples.functions![index].function.name;
+  dynamic get _function {
+    if (isCodeTree) {
+      return samples.codes[index].code!.function!;
+    }
+    return samples.functions![index].function;
+  }
+
+  String? get name {
+    if (isCodeTree) {
+      return samples.codes[index].code!.name;
```

Comment:
should we add a helper `get _code => samples.codes[index].code;` since this is used in a couple places?